Multiplatform/Collections/Artists/ArtistList.swift

```diff
@@ -11,13 +11,78 @@ import AFBase
 struct ArtistList: View {
     let artists: [Artist]
     
+    var count = 0
+    var loadMore: LoadCallback = nil
+    
+    @State private var working = false
+    @State private var search: String = ""
+    
     var body: some View {
+        let filtered = filter(artists: artists)
         List {
-            ForEach(artists) {
-                ArtistListRow(artist: $0)
+            ForEach(filtered) { artist in
+                ModifiedArtistListRow(artist: artist, expand: artist == filtered.last ? expand : nil)
+            }
+            ForEach(0..<(max(0, count - artists.count)), id: \.hashValue) { _ in
+                ArtistListRow.placeholder
+                    .onAppear { expand() }
             }
         }
         .listStyle(.plain)
+        .searchable(text: $search, placement: .navigationBarDrawer(displayMode: .automatic), prompt: "search.artists")
```

Comment:
I think the searchable modifier and the `expand` function should be moved to `ArtistsView`. The `TrackList` component is quite complex and does a lot of stuff, but simpler things like this or the `AlbumsGrid` shouldn't have to handle the search.

Although there is an argument to be made that the `TrackList` shouldn't contain the search functionality either.